File: AdisyonUygulamaJavaXMLsiz/app/src/main/java/com/example/adisyonuygulamajava/anaekran/MasaCard.java
package com.example.adisyonuygulamajava.anaekran;

import android.content.Context;
import android.graphics.Color;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.cardview.widget.CardView;

import com.example.adisyonuygulamajava.model.Masa;
import com.example.adisyonuygulamajava.viewmodel.UseCases;

public class MasaCard {

    public interface OnMasaClickListener {
        void onClick(Masa masa);
    }

    private final CardView cardView;
    private final TextView masaAdiTV;
    private final TextView masaFiyatTV;
    private final TextView masaSureTV;
    private Masa masa;

    public MasaCard(Context context, OnMasaClickListener listener) {
        cardView = new CardView(context);
        masaAdiTV = new TextView(context);
        masaFiyatTV = new TextView(context);
        masaSureTV = new TextView(context);

        // CardView özellikleri
        cardView.setRadius(24f);
        cardView.setCardElevation(8f);
        cardView.setUseCompatPadding(true);

        ViewGroup.MarginLayoutParams layoutParams = new ViewGroup.MarginLayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.WRAP_CONTENT
        );
        layoutParams.setMargins(20, 20, 20, 20);
        cardView.setLayoutParams(layoutParams);

        // Dikey LinearLayout
        LinearLayout verticalLayout = new LinearLayout(context);
        verticalLayout.setOrientation(LinearLayout.VERTICAL);
        verticalLayout.setPadding(30, 30, 30, 30);

        masaAdiTV.setTextSize(20f);
        masaAdiTV.setTextColor(Color.BLACK);

        masaFiyatTV.setTextSize(18f);
        masaFiyatTV.setTextColor(Color.DKGRAY);

        masaSureTV.setTextSize(16f);
        masaSureTV.setTextColor(Color.GRAY);

        verticalLayout.addView(masaAdiTV);
        verticalLayout.addView(masaFiyatTV);
        verticalLayout.addView(masaSureTV);

        cardView.addView(verticalLayout);

        // Tıklama olayı
        cardView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (masa != null) {
                    listener.onClick(masa);
                }
            }
        });
    }

    public CardView getCardView() {
        return cardView;
    }

    public void setData(Masa masa) {
        this.masa = masa;
        masaAdiTV.setText("Masa " + masa.getId());
        masaFiyatTV.setText("Tutar: " + UseCases.fiyatYaz(masa.getToplam_fiyat()));
        masaSureTV.setText("Süre: " + masa.getSure());
    }

    public void setBackgroundColor(int color) {
        cardView.setCardBackgroundColor(color);
    }

    // Kotlin'deki fiyatYaz uzantısına karşılık Java fonksiyonu
    private String fiyatYaz(float fiyat) {
        return String.format("%.2f ₺", fiyat);
    }
}

File: AdisyonUygulamaKotlinXMLsiz/app/src/main/java/com/example/adisyonuygulamakotlin/view/anaekran/MasaCard.kt
package com.example.adisyonuygulamakotlin.view.anaekran

import android.content.Context
import android.graphics.Color
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import androidx.cardview.widget.CardView
import com.example.adisyonuygulamakotlin.model.Masa
import com.example.adisyonuygulamakotlin.utils.SizeType
import com.example.adisyonuygulamakotlin.utils.fiyatYaz
import com.example.adisyonuygulamakotlin.utils.marginEkle
import com.example.adisyonuygulamakotlin.utils.paddingEkle
import com.example.adisyonuygulamakotlin.utils.sizeBelirle

class MasaCard(
    private val context: Context,
    private val onClick: (Masa) -> Unit
) {

    private val cardView = CardView(context)
    private val masaAdiTV = TextView(context)
    private val masaFiyatTV = TextView(context)
    private val masaSureTV = TextView(context)

    private var masa: Masa? = null

    init {
        cardView.apply {
            radius = 24f
            cardElevation = 8f
            useCompatPadding = true
            layoutParams = ViewGroup.MarginLayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.WRAP_CONTENT
            ).apply {
                setMargins(20, 20, 20, 20)
            }
        }

        val verticalLayout = LinearLayout(context).apply {
            orientation = LinearLayout.VERTICAL
            setPadding(30, 30, 30, 30)
            addView(masaAdiTV.apply {
                textSize = 20f
                setTextColor(Color.BLACK)
            })
            addView(masaFiyatTV.apply {
                textSize = 18f
                setTextColor(Color.DKGRAY)
            })
            addView(masaSureTV.apply {
                textSize = 16f
                setTextColor(Color.GRAY)
            })
        }

        cardView.addView(verticalLayout)

        cardView.setOnClickListener {
            masa?.let { onClick(it) }
        }
    }

    fun cardView(): CardView = cardView

    fun setData(masa: Masa) {
        this.masa = masa
        masaAdiTV.text = "Masa ${masa.id}"
        masaFiyatTV.text = "Tutar: ${masa.toplam_fiyat.fiyatYaz()}"
        masaSureTV.text = "Süre: ${masa.sure}"
    }

    fun setBackgroundColor(color: Int) {
        cardView.setCardBackgroundColor(color)
    }
}


